src/lib.rs

```diff
@@ -297,3 +297,17 @@ broker_handler!(CallbackHandler, |app, req| {
         try!(oidc::verify(app, session, code))
     )
 });
+
+
+#[cfg(test)]
+mod tests {
+    // TODO: Refactor so these tests fit somewhere more natural...
```

Comment:
This is more of a test of `emailaddress` itself, so I'm not sure it belongs here at all.

Tests here could call the handler functions and check the result, but that also sounds more like integration testing than unit testing. Stuff like `crypto.rs` would be good unit test candidates, I think.